tasks/main.yml

```diff
@@ -9,33 +9,52 @@
 
 - name: precondition - java_subversion
   fail: msg="ERROR - required variable 'java_subversion' missing."
-  when: java_subversion is not defined
+  when: java_version is regex("\d+\.\d+\.\d+") and java_subversion is not defined
```

Comment:
I think this should be `java_version is not regex(...)`, since if it's a semver, then subversion does not need to be defined.

Reply:
oops, right